ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/fillers/MaxLoopFlowFiller.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.farao_community.farao.linear_rao.fillers;
+
+import com.farao_community.farao.commons.FaraoException;
+import com.farao_community.farao.data.crac_api.Cnec;
+import com.farao_community.farao.data.crac_api.Crac;
+import com.farao_community.farao.data.crac_loopflow_extension.CnecLoopFlowExtension;
+import com.farao_community.farao.data.crac_loopflow_extension.CracLoopFlowExtension;
+import com.farao_community.farao.flowbased_computation.impl.LoopFlowComputation;
+import com.farao_community.farao.linear_rao.AbstractProblemFiller;
+import com.farao_community.farao.linear_rao.LinearRaoData;
+import com.farao_community.farao.linear_rao.LinearRaoProblem;
+import com.google.ortools.linearsolver.MPConstraint;
+import com.google.ortools.linearsolver.MPVariable;
+
+import java.util.*;
+
+/**
+ * @author Pengbo Wang {@literal <pengbo.wang at rte-international.com>}
+ */
+public class MaxLoopFlowFiller extends AbstractProblemFiller {
+
+    private Set<Cnec> preventiveCnecs; //currently we only forcus on preventive state cnec
+    private CracLoopFlowExtension cracLoopFlowExtension;
+
+    public MaxLoopFlowFiller(LinearRaoProblem linearRaoProblem, LinearRaoData linearRaoData) {
+        super(linearRaoProblem, linearRaoData);
+        Crac crac = linearRaoData.getCrac();
+        this.preventiveCnecs = crac.getCnecs(crac.getPreventiveState());
+        this.cracLoopFlowExtension = crac.getExtension(CracLoopFlowExtension.class);
+    }
+
+    @Override
+    public void fill() {
+        buildMaxLoopFlowConstraint();
+    }
+
+    /**
+     * Loopflow F_(0,all)_current = flowVariable - PTDF * NetPosition, where flowVariable is a MPVariable in linearRao
+     * then max loopflow MPConstraint is: - maxLoopFlow <= currentLoopFlow <= maxLoopFlow
+     * we define loopFlowShift = PTDF * NetPosition, then
+     * -maxLoopFlow + loopFlowShift <= flowVariable <= maxLoopFlow + loopFlowShift,
+     */
+    private void buildMaxLoopFlowConstraint() {
+        LoopFlowComputation loopFlowComputation = new LoopFlowComputation(linearRaoData.getCrac(), cracLoopFlowExtension);
+        Map<Cnec, Map<String, Double>> ptdfResults = loopFlowComputation.computePtdfOnCurrentNetwork(linearRaoData.getNetwork()); // get ptdf
+        Map<String, Double> referenceNetPositionByCountry = loopFlowComputation.getRefNetPositionByCountry(linearRaoData.getNetwork()); // get Net positions
+        Map<Cnec, Double> loopFlowShifts = loopFlowComputation.buildZeroBalanceFlowShift(ptdfResults, referenceNetPositionByCountry); //compute PTDF * NetPosition
```

Comment:
I would refactor the LoopFlowComputation class to be able to directly call the buildZeroBalanceFlowShift with a network. (You could simply overload the current buildZeroBalanceFlowShift method and put these few steps in it)

Reply:
OK